desktop/core/src/desktop/js/utils/hueAnalytics.ts

```diff
@@ -14,18 +14,73 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { getLastKnownConfig } from '../config/hueConfig';
 import { post } from '../api/utils';
 import { hueWindow } from '../types/types';
 
-export default {
-  log(app: string, page: string): void {
-    if (typeof (<hueWindow>window).trackOnGA == 'function') {
-      (<hueWindow>window).trackOnGA!(app + '/' + page);
+// HOW TO TRACK EVENTS
+// a) Calling the hueAnalytics.log or hueAnalytics.convert manually
+// b) Using the attribute "data-hue-analytics" on a clickable HTML element
+// and set a string value containing the area followed by a colon and the action e.g.
+// <a href="#" data-hue-analytics="metastore:remove stuff">Remove</a>
+//
+// Setting the additional attribute data-hue-analytics-prio="true" will cause the interaction
+// to be logged in the Hue backend, independent of the collect_usage setting. This should be used
+// with caution to prevent flooding the log.
+
+const formatGaData = (
+  area: string,
+  action: string
+): { type: string; name: string; params: { action: string; version: string } } => ({
+  type: 'event',
+  name: area,
+  params: {
+    action,
+    version: (<hueWindow>window).HUE_VERSION || ''
+  }
+});
+
+export const hueAnalytics = {
+  log(area: string, action: string, isPrioritised?: boolean): void {
+    const config = getLastKnownConfig();
+    const typedWindow = <hueWindow>window;
+    if (isPrioritised) {
+      this.convert(area, action);
+    }
+
+    if (config?.hue_config?.collect_usage) {
+      const { type, name, params } = formatGaData(area, action);
+
+      // Quick debug mode to check that the analytics logging is working when developing new features
+      // Turn on by typing "window.hueDebugAnalytics = true;" in the browser JS console or setting
+      // "window.hueDebugAnalytics = true" in hue.js;
+      if (typedWindow.DEV && typedWindow.hueDebugAnalytics) {
+        console.info('Analytics debug:', type, name, params);
+      }
+
+      if (!typedWindow.DEV) {
+        typedWindow.gtag?.(type, name, params);
```

Comment:
Didn't know about the dot in gtag.?, I always found it annoying that ?(  didn't work, thanks! :-)

Reply:
I actually learned it myself when working on this Jira :-)